get_next_line_utils.c

```diff
@@ -62,34 +62,48 @@ char	*ft_strndup(const char *s, size_t n)
 	ret = malloc(n + 1);
 	if (!ret)
 		return (NULL);
-	ft_strlcpy(ret, s, n + 1);
+	gnl_strlcpy(ret, s, n + 1);
 	return (ret);
 }
 
-char	*ft_strjoin_with_free(const char *s1, const char *s2, int free_flag)
+char	*gnl_strjoin_with_free(char **s1, char **s2, int free_flag)
 {
 	char	*ret;
 	size_t	s1_len;
 	size_t	s2_len;
 
-	s1_len = ft_strlen(s1);
-	s2_len = ft_strlen(s2);
-	if (!s1 && !s2)
+	if (!*s1 && !*s2)
 		return (NULL);
+	s1_len = gnl_strlen(*s1);
+	s2_len = gnl_strlen(*s2);
 	ret = malloc(s1_len + s2_len + 1);
-	if (ret && s1)
-		ft_strlcpy(ret, s1, s1_len + 1);
-	if (ret && s2)
-		ft_strlcpy(ret + s1_len, s2, s2_len + 1);
-	if (s1 && free_flag & 0b10)
+	if (!ret)
+	{
+		if (free_flag & 0b10)
+		{
+			free(*s1);
+			*s1 = NULL;
+		}
+		if (free_flag & 0b01)
+		{
+			free(*s2);
+			*s2 = NULL;
+		}
+		return (NULL);
+	}
```

Comment:
I think this if statement is not necessary. because `if (ret && *s1)` and `if (ret && *s2)` include NULL checking.

Reply:
Confirmed. You are right.